db/kv_dense_channel_storage_test.go

```diff
@@ -238,6 +238,84 @@ func TestDenseBlockRemovalByKey(t *testing.T) {
 
 }
 
+func TestDenseBlockRollbackTo(t *testing.T) {
+	base.EnableLogKey("ChannelStorage")
+	base.EnableLogKey("ChannelStorage+")
+	indexBucket := testIndexBucket()
+	defer indexBucket.Close()
+
+	block := NewDenseBlock("block1", nil)
+
+	// Inserts the following entries:
+	// [0,1] [1,2] [2,3] [0,4] [1,5] [2,6] [0,7] [1,8] [2,9] [0,10]
+	entries := make([]*LogEntry, 10)
+	for i := 0; i < 10; i++ {
+		sequence := i + 1
+		vbNo := i % 3 // mix up the vbuckets
+		entries[i] = makeBlockEntry(fmt.Sprintf("doc%d", i), "1-abc", vbNo, sequence, IsNotRemoval, IsAdded)
+	}
+	overflow, pendingRemoval, _, err := block.AddEntrySet(entries, indexBucket)
+	assertNoError(t, err, "Error adding entry set")
+	assert.Equals(t, len(overflow), 0)
+	assert.Equals(t, len(pendingRemoval), 0)
+	assert.Equals(t, block.getEntryCount(), uint16(10))
+
+	foundEntries := block.GetAllEntries()
+	assert.Equals(t, len(foundEntries), 10)
+	assertLogEntry(t, foundEntries[0], "doc0", "1-abc", 0, 1)
+	assertLogEntry(t, foundEntries[1], "doc1", "1-abc", 1, 2)
+	assertLogEntry(t, foundEntries[2], "doc2", "1-abc", 2, 3)
+	assertLogEntry(t, foundEntries[3], "doc3", "1-abc", 0, 4)
+	assertLogEntry(t, foundEntries[4], "doc4", "1-abc", 1, 5)
+	assertLogEntry(t, foundEntries[5], "doc5", "1-abc", 2, 6)
+	assertLogEntry(t, foundEntries[6], "doc6", "1-abc", 0, 7)
+	assertLogEntry(t, foundEntries[7], "doc7", "1-abc", 1, 8)
+	assertLogEntry(t, foundEntries[8], "doc8", "1-abc", 2, 9)
+	assertLogEntry(t, foundEntries[9], "doc9", "1-abc", 0, 10)
+
+	// Rollback should complete in this block
+	rollbackComplete, err := block.RollbackTo(2, 5, indexBucket)
+	assertNoError(t, err, "Error rolling back")
+	assert.Equals(t, rollbackComplete, true)
+	assert.Equals(t, block.getEntryCount(), uint16(8))
+
+	foundEntries = block.GetAllEntries()
+	assert.Equals(t, len(foundEntries), 8)
+	assertLogEntry(t, foundEntries[0], "doc0", "1-abc", 0, 1)
+	assertLogEntry(t, foundEntries[1], "doc1", "1-abc", 1, 2)
+	assertLogEntry(t, foundEntries[2], "doc2", "1-abc", 2, 3)
+	assertLogEntry(t, foundEntries[3], "doc3", "1-abc", 0, 4)
+	assertLogEntry(t, foundEntries[4], "doc4", "1-abc", 1, 5)
+	assertLogEntry(t, foundEntries[5], "doc6", "1-abc", 0, 7)
+	assertLogEntry(t, foundEntries[6], "doc7", "1-abc", 1, 8)
+	assertLogEntry(t, foundEntries[7], "doc9", "1-abc", 0, 10)
+
+	// Rollback should NOT complete in this block (haven't seen a sequence earlier than 1 in vb 1)
+	rollbackComplete, err = block.RollbackTo(1, 1, indexBucket)
```

Comment:
I'm a little confused on why the rollback didn't complete.  How could you rollback vb 1 such that it removed all of it's log entries? `block.RollbackTo(1, 0, indexBucket)`?  Also what would have happened if  `block.RollbackTo(1, 2, indexBucket)` was called here?